### ISSUE TITLE
No error with misspelled output `steps.cache.outputs.primary-key`

### ISSUE THREAD

**Issue by andreasabel** [rhysd/actionlint#365]
Thanks for `actionlint`, made my life so much better!

Here is problem with checking the `outputs` of a step:
The `actions/cache` action has output `cache-primary-key`, but I misspelled it as `primary-key`, not getting a warning from `actionlint`.
```yaml
    - uses: actions/cache/restore@v3
      id:   cache
      with:
        path: ...
        key: ...
        restore-keys: ...
...
    -  if:   ${{ !steps.cache.outputs.cache-hit }}
      uses: actions/cache/save@v3
      with:
        path: ...
        key: ${{ steps.cache.outputs.primary-key }}
```
Can `actionlint` track the valid outputs of a previous step?
